pkgs/development/python-modules/hcipy/default.nix

```
@@ -0,0 +1,75 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  pythonOlder,
+
+  # build-system
+  setuptools-scm,
+
+  # dependencies
+  numpy,
+  scipy,
+  matplotlib,
+  pillow,
+  pyyaml,
+  astropy,
+  imageio,
+  xxhash,
+  numexpr,
+  asdf,
+  importlib-resources,
+
+  # tests
+  pytest,
+  pytest-cov,
+  pytestCheckHook,
+  mpmath,
+  dill,
+  flake8,
+}:
+
+buildPythonPackage rec {
+  pname = "hcipy";
+  version = "0.6.0";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-Kt2T6EgCrPTJGiygjmFUh1HH1DthFcZ2mMlnlu0HxsE=";
+  };
+
+  build-system = [ setuptools-scm ] ++ lib.optionals (pythonOlder "3.9") [ importlib-resources ];
```

```suggestion
  build-system = [ setuptools-scm ];
```
Oldest Python in nixpkgs is 3.9, and it doesn't even build on that version due to dependencies.

---

Done. I also removed the now obsolete input parameters.